Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1778 @@
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $KeyVaultName,
+    [Parameter(Mandatory = $true)]
+    [string[]]
+    $PublicIpAddresses
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  ForEach ($publicIpAddress in $PublicIpAddresses)
+  {
+    Set-KeyVaultPublicNetworkAccessEnabledForIpAddress `
+      -SubscriptionId $SubscriptionId `
+      -ResourceGroupName $ResourceGroupName `
+      -KeyVaultName $KeyVaultName `
+      -PublicIpAddress $publicIpAddress
+  }
+}
+
+function Set-KeyVaultSecurePublicNetworkSettings()
+{
+  <#
+    .SYNOPSIS
+    This command updates an existing Key Vault to enable public network access with default action Deny, and to allow trusted Azure services.
+    .DESCRIPTION
+    This command updates an existing Key Vault to enable public network access with default action Deny, and to allow trusted Azure services. This corresponds to the "Allow public access from specific virtual network and IP addresses" setting in the Azure portal. Use this to configure the Key Vault's overall network settings apart from individual rules.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Key Vault.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Key Vault.
+    .PARAMETER KeyVaultName
+    The Key Vault name.
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Set-KeyVaultSecurePublicNetworkSettings -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -KeyVaultName "MyKeyVaultName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $KeyVaultName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $apiVersion = "2022-07-01"
+
+  $payload = `
+  '
+  {
+    "properties": {
+      "networkAcls": {
+      "defaultAction": "Deny",
+      "bypass": "AzureServices"
+      },
+      "publicNetworkAccess": "Enabled"
+    }
+  }
+  '
+
+  $result = Invoke-AzRestMethod `
+    -Subscription $SubscriptionId `
+    -ResourceGroupName $ResourceGroupName `
+    -ResourceProviderName 'Microsoft.KeyVault' `
+    -ResourceType 'vaults' `
+    -Name $KeyVaultName `
+    -ApiVersion $apiVersion `
+    -Method PATCH `
+    -Payload $payload
+
+    Write-InformationFormatted -MessageData "Result status code was " + $result.StatusCode -ForegroundColor Cyan
+
+    if ("200" -ne $result.StatusCode)
+    {
+      $result
+    }
```

Comment:
Suggestion: Logging that operation was successful in green might be helpful for end user.